File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution(object):
    def removeDuplicates(self, nums):
        n = len(nums)
        j = 0
        for i in range(0 , n):
            if j  < 2 or nums[i] != nums[j-2]:
                nums[j] = nums[i]
                j += 1
        return j

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int i = 0  , j = 0;
        int cnt = 0;
        int val = nums[0];
        for(int i = 0 ; i < n ; i++){
            if(nums[i] == val){
                cnt++;
            }else{
                if(cnt >= 2){
                    nums[j++] = val;
                    nums[j++] = val;
                }else if(cnt == 1){
                    nums[j++] = val;
                }
                cnt = 1;
                val = nums[i];
            }
        }
        if(cnt >= 2){
            nums[j++] = val;
            nums[j++] = val;
        }else if(cnt == 1){
            nums[j++] = val;
        }
        return j;
    }
};